alpha/action/migrations/migrations.go

```diff
@@ -0,0 +1,122 @@
+package migrations
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+type MigrationToken string
+
+const MigrationTokenInvalid MigrationToken = ""
+
+type Migration interface {
+	Token() MigrationToken
+	Name() string
+	Help() string
+	Migrate(*declcfg.DeclarativeConfig) error
+}
+
+func newMigration(name string, help string, fn func(config *declcfg.DeclarativeConfig) error) Migration {
+	s := strings.ReplaceAll(help, "`", "\"")
+	return &simpleMigration{token: MigrationToken(name), name: name, help: s, fn: fn}
+}
+
+type simpleMigration struct {
+	token MigrationToken
+	name  string
+	help  string
+	fn    func(*declcfg.DeclarativeConfig) error
+}
+
+func (s simpleMigration) Token() MigrationToken {
+	return s.token
+}
+
+func (s simpleMigration) Name() string {
+	return s.name
+}
+
+func (s simpleMigration) Migrate(config *declcfg.DeclarativeConfig) error {
+	return s.fn(config)
+}
+
+func (s simpleMigration) Help() string {
+	return s.help
+}
+
+type Migrations struct {
+	Migrations []Migration
+}
+
+func GetLastMigration() MigrationToken {
+	if len(allMigrations) == 0 {
+		return ""
+	}
+	return allMigrations[len(allMigrations)-1].Token()
+}
+
+func ValidateName(name string) (MigrationToken, error) {
+	for _, migration := range allMigrations {
+		if migration.Name() == name {
+			return migration.Token(), nil
+		}
+	}
+	return MigrationTokenInvalid, fmt.Errorf("unknown migration level %q", name)
+}
```

Comment:
I feel like we should be able to incorporate this function into `NewMigrations` (or make it unexported and call it from there), especially if you agree that it makes sense for the `Render` struct (and others) to accept a *migrations.Migrations` directly, as I suggested elsewhere.